make/common/JarArchive.gmk

```diff
@@ -212,25 +212,30 @@ define SetupJarArchiveBody
             | $(SED) 's|$$(src)/|-C $$(src) |g' >> \
         $$($1_BIN)/_the.$$($1_JARNAME)_contents) $$(NEWLINE) )
   endif
-  $1_SUPDATE_CONTENTS=$$($1_JAR_CMD) $$($1_JAR_UPDATE_OPTIONS) $$@ @$$($1_BIN)/_the.$$($1_JARNAME)_contents $$(NEWLINE)
+  $1_SUPDATE_CONTENTS=$$($1_JAR_CMD) --update $$($1_JAR_OPTIONS) --file $$@ @$$($1_BIN)/_the.$$($1_JARNAME)_contents $$(NEWLINE)
 
   # Use a slightly shorter name for logging, but with enough path to identify this jar.
   $1_NAME:=$$(subst $$(OUTPUTDIR)/,,$$($1_JAR))
 
+  # If reproducible build and the boot jdk jar supports --date
+  # then specify the --date using SOURCE_DATE in ISO-8601
+  $1_JAR_SOURCE_DATE :=
+  ifeq ($$(ENABLE_REPRODUCIBLE_BUILD), true)
+    ifeq ($$(BOOT_JDK_JAR_SUPPORTS_DATE), true)
+      $1_JAR_SOURCE_DATE := --date $(SOURCE_DATE_ISO_8601)
+    endif
+  endif
   ifneq (,$$($1_CHECK_COMPRESS_JAR))
-    $1_JAR_CREATE_OPTIONS := c0fm
-    $1_JAR_UPDATE_OPTIONS := u0f
+    $1_JAR_OPTIONS := --no-compress $$($1_JAR_SOURCE_DATE)
     ifeq ($(COMPRESS_JARS), true)
-      $1_JAR_CREATE_OPTIONS := cfm
-      $1_JAR_UPDATE_OPTIONS := uf
+      $1_JAR_OPTIONS := $$($1_JAR_SOURCE_DATE)
     endif
   else
-    $1_JAR_CREATE_OPTIONS := cfm
-    $1_JAR_UPDATE_OPTIONS := uf
+    $1_JAR_OPTIONS := $$($1_JAR_SOURCE_DATE)
```

Comment:
Instead of adding $$($1_JAR_SOURCE_DATE) to every branch of this conditional, you can use += to build the $1_JAR_OPTIONS variable one step at a time. I don't think we need a separate variable JAR_SOURCE_DATE.

Reply:
reduced as stated